src/labone/nodetree/node.py

```diff
@@ -0,0 +1,863 @@
+        The node itself is also not counted as its child.
+
+        Args:
+            child_node: Potential child node
+
+        Returns:
+            Boolean if passed node is a child node
+        """
+        return (self.path_segments == child_node._path_segments[:-1] and
+                    child_node.path_segments[-1] in self._subtree_structure.keys())
+
+class ResultNode(MetaNode):
+    def __init__(self,
+                 path_segments: t.Tuple[t.Union[str, int]],
+                 subtree_structure,
+                 value_structure: TreeProp,
+                 path_aliases):
+
+        super().__init__(path_segments, subtree_structure, path_aliases)
+        self._value_structure: TreeProp = value_structure
+
+    def try_generate_subnode(self, next_path_segment: NormalizedPathSegment):
+        extended_path = (*self._path_segments, next_path_segment)
+
+        try:
+            next_subtree = self._subtree_structure[next_path_segment]
+        except KeyError:
+            if next_path_segment == WILDCARD:
+                raise RuntimeError(f'Wildcards * All are not allowed in a tree '
+                                   f'representing measurement results')
+
+            raise RuntimeError(
+                f"node path_segments '{self}' cannot be extended with /{next_path_segment}\n"
+                f"viable extentions would be {list(self._subtree_structure.keys())}")
+        else:
+            deeper_node = ResultNode(extended_path,
+                              self._subtree_structure[next_path_segment],
+                              self._value_structure,
+                              self._path_aliases)
+
+            if next_subtree == {}:
+                # give value instead of subnode if already at a leaf
+                # generate hypothetical node in order to apply normal behaviour,
+                # including path alias redirection
+                return self._value_structure.get(deeper_node) #.get_with_segments(extended_path)
+
+            return deeper_node
+
+    def __dir__(self) -> t.Iterable[str]:
+        return [pythonify_path_segment(p) for p in self._subtree_structure.keys()]
+
+    def __call__(self, *args, **kwargs):
+        raise RuntimeError('Trying to get/set on a result node, which is not possible. This node only represents '
+                           'values of a former get-request. To interact with a device, '
+                           'make sure to operate on normal nodes.')
+
+
+class Node(MetaNode):
+    _matches_index = UniqueCounter()
+
+    def __init__(self,
+                 tree_manager: NodeTreeManager,
+                 path_segments: t.Tuple[t.Union[str, int]],
+                 subtree_structure: NestedDict | None = None,
+                 path_aliases: dict[tuple[NormalizedPathSegment], tuple[NormalizedPathSegment]] | None = None
+                 ):
+
+        if path_aliases is None:
+            path_aliases = {}
+        if subtree_structure is None:
```

Comment:
Make `subtree_structure` not optional and construct it if necessary in the calling method